ci-generate-sync.js

```diff
@@ -0,0 +1,97 @@
+const fs = require("fs");
+
+const services = [
+  "auditlogs",
+  "create-artifact-worker",
+  "deployments",
+  "deviceauth",
+  "deviceconfig",
+  "deviceconnect",
+  "devicemonitor",
+  "inventory",
+  "mender-gateway",
+  "tenantadm",
+  "useradm",
+  "workflows",
+];
+
+const pipelineTemplate = (jobs, shared) => `stages:
+  - version-bump
+${shared}
+${jobs.join("")}
+`;
+
+const getChangesRule = ({
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `- changes:
+        paths: ["backend/services/${service}/**/*", "backend/pkg/**/*"]
+        compare_to: "${RULES_CHANGES_COMPARE_TO_REF}"`;
+
+const sharedTemplate = ({ CI_COMMIT_REF_NAME, CI_COMMIT_SHA }) => `
+.trigger:mender-helm-version-bump:
+  stage: version-bump
+  allow_failure: true
+  trigger:
+    project: Northern.tech/Mender/mender-helm
+    strategy: depend
+  resource_group: mender-helm
+  variables:
+    SYNC_IMAGE_TAG: $DOCKER_PUBLISH_COMMIT_TAG
+    DOCKER_PUBLISH_COMMIT_TAG: ${CI_COMMIT_REF_NAME}_${CI_COMMIT_SHA}
+    GITHUB_HELM_REPO: "github.com/mendersoftware/mender-helm.git"
+    CHART_DIR: "mender"
+`;
+
+const template = ({
+  CI_COMMIT_REF_NAME,
+  CI_PIPELINE_ID,
+  CONTAINER_NAME,
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `
+trigger:mender-helm-version-bump:${service}:staging:
+  extends:
+    - .trigger:mender-helm-version-bump
+  rules:
+    ${getChangesRule({ RULES_CHANGES_COMPARE_TO_REF, service })}
+    - if: $CI_COMMIT_BRANCH =~ /^(staging)$/
```

Comment:
Maybe this is a topic for the QA meeting, but IMO we could drop the staging branch and deploy to staging from main.
```suggestion
  rules:
    ${getChangesRule({ "HEAD~1", service })}
    - if: $CI_COMMIT_BRANCH == main
```